tests/Api/Ip.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * Copyright (C) 2013 Mailgun
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+namespace Mailgun\Tests\Api;
+
+use GuzzleHttp\Psr7\Response;
+use Mailgun\Api\Ip;
+use Mailgun\Model\Ip\IndexResponse;
+
+class IpTest extends TestCase
+{
+    protected function getApiClass()
+    {
+        return Ip::class;
+    }
+
+    public function testIndexAll()
+    {
+        $this->setRequestMethod('GET');
+        $this->setRequestUri('/v3/ips');
+        $this->setHttpResponse(new Response(200, ['Content-Type' => 'application/json'], <<<'JSON'
+{
+  "items": ["192.161.0.1", "192.168.0.2"],
+  "total_count": 2
+}
+JSON
+        ));
+
+        $api = $this->getApiInstance();
+        /** @var IndexResponse $response */
+        $response = $api->index(null);
+        $this->assertInstanceOf(IndexResponse::class, $response);
+        $this->assertEquals(2, $response->getTotalCount());
+        $this->assertEquals('192.161.0.1', $response->getItems()[0]);
+        $this->assertEquals('192.168.0.2', $response->getItems()[1]);
+    }
+    
+    public function testIndexOnlyDedicated()
+    {
+        $this->setRequestMethod('GET');
+        $this->setRequestUri('/v3/ips?dedicated=1');
```

Comment:
I am not sure about this one. According to the docs, we are supposed to pass `true` instead of `1`

https://documentation.mailgun.com/en/latest/api-ips.html#ips

![image](https://user-images.githubusercontent.com/7809429/95360579-79339280-08c3-11eb-8511-f74ff5eaedd6.png)

Moreover, as part of the example, we are supposed to pass that `true` value as a string instead of an integer or boolean (at least, as defined for the `curl` call):

https://documentation.mailgun.com/en/latest/api-ips.html#example

![image](https://user-images.githubusercontent.com/7809429/95360635-8e102600-08c3-11eb-9c26-0821fd9325ca.png)

Are we 100% sure this is the case? I would prefer not merging this until we can confirm it won't break the implementation.

Reply:
I ran the following commands (I edited the results to remove the API key):

```
$ curl -s --user 'api:API_KEY' -G https://api.mailgun.net/v3/ips?dedicated=false
{"assignable_to_pools":["159.135.238.30"],"items":["209.61.151.242"],"total_count":1}

$ curl -s --user 'api:API_KEY' -G https://api.mailgun.net/v3/ips?dedicated=true
{"assignable_to_pools":["159.135.238.30"],"items":["159.135.225.45","166.78.69.72","159.135.238.30"],"total_count":3}

$ curl -s --user 'api:API_KEY' -G https://api.mailgun.net/v3/ips
{"assignable_to_pools":["159.135.238.30"],"items":["159.135.225.45","166.78.69.72","159.135.238.30","209.61.151.242"],"total_count":4}

$ curl -s --user 'api:API_KEY' -G https://api.mailgun.net/v3/ips?dedicated=0
{"assignable_to_pools":["159.135.238.30"],"items":["209.61.151.242"],"total_count":1}

$ curl -s --user 'api:API_KEY' -G https://api.mailgun.net/v3/ips?dedicated=1
{"assignable_to_pools":["159.135.238.30"],"items":["159.135.238.30","159.135.225.45","166.78.69.72"],"total_count":3}
```

It appears that:
1. `0`/`1` return the same results as `false`/`true` for the dedicated parameter
2. Without specifying the parameter `dedicated` returns different results than either `true`/`false`/`0`/`1`.
3. There is an additional key in the response `assignable_to_pools` which doesn't appear in the documentation

Reply:
:+1: for points 1 and 2

Maybe I would suggest, If you don't mind, adding the mapping for the property that we don't have at the current stage, so that we can cover this case too?

Reply:
I added `assignable_to_pools` and pushed a change. See commit 8c092ab5952b82a2207cc251345483082d3ded4f